app/uk/gov/hmrc/agentepayeregistration/validators/AgentEpayeRegistrationValidator.scala

```diff
@@ -0,0 +1,98 @@
+package uk.gov.hmrc.agentepayeregistration.validators
+
+import cats.Semigroup
+import cats.data.Validated
+import cats.data.Validated.{Invalid, Valid}
+import uk.gov.hmrc.agentepayeregistration.models.{Failure, RegistrationRequest}
+
+object ValidatedSemigroup {
+  implicit def validatedSemigroup[A] = new Semigroup[Validated[Failure, Unit]] {
+    def combine(x: Validated[Failure, Unit], y: Validated[Failure, Unit]): Validated[Failure, Unit] = (x, y) match {
```

Comment:
In this case, we are implementing a ```Semigroup``` for validation which allows us to perform various operations supporting Semigroup. I think ```|@|``` just combines all the results from ```Validated``` together only.